src/drivers/uavcan/CMakeLists.txt

```diff
@@ -46,6 +46,11 @@ if(CONFIG_ARCH_CHIP)
 	elseif(${CONFIG_ARCH_CHIP} MATCHES "stm32h7")
 		set(UAVCAN_DRIVER "stm32h7")
 		set(UAVCAN_TIMER 5) # The default timer is TIM5
+		if(CONFIG_CDCACM_VENDORSTR)
+			if(${CONFIG_CDCACM_VENDORSTR} MATCHES "CUAV")
+				set(UAVCAN_TIMER 2)
```

Comment:
Looks good other than this part. The goal is to not have any vendor specific code outside of the board directory. We should add something to the board (boards/cuav/x7pro/default.cmake) that sets this.

Alternatively we could take a quick look at the other H7 boards to see if we can get away with changing the default timer on stm32h7.

Reply:
![image](https://user-images.githubusercontent.com/30532769/87895792-e67c1800-ca78-11ea-8f5a-868db188115b.png)
Can I define it this way?

Reply:
Yes that works for now.

Later it might make sense to move this and UAVCAN_INTERFACES to `board_config.h` and out of cmake entirely.

Reply:
What do I need to do?

Reply:
@dagar Hello, how is the progress?

Reply:
@dagar Do I need to modify it?